File: 08/ts/src/challenge.ts
import fs from "node:fs";
import { magicSolvePart2, parseChallenge, parseChallengeOptimized, powerSolvePart2, solvePart2HeavilyOptimized } from "./utils.ts";

const fileName = Deno.args[0]; // first argument
console.log(`Reading ${fileName}`);

const input = fs.readFileSync(fileName, { encoding: "utf-8" });

const walker = parseChallenge(input);
const powerWalker = parseChallengeOptimized(input);

// const resPart1 = solve(parsedInput);
// const resPart2 = solvePart2HeavilyOptimized(walker);
// const resMagic = magicSolvePart2(parsedInput)
const resPower = powerSolvePart2(powerWalker);

// console.log("Part 1's result is", resPart1);
// console.log("Part 2's result is", resPart2);
// console.log("Part 2's result is", resMagic);
console.log("Part 2's result is", resPower);

File: 08/python/src/challenge.py
import sys

def read_file(file_name: str) -> str:
    with open(file_name, 'r', encoding='utf-8') as file:
        return file.read()

# Assuming the utility functions are in a module named utils
from utils import parse_challenge, solve

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Please provide the input file name as an argument.")
        sys.exit(1)
    
    file_name = sys.argv[1]
    print(f"Reading {file_name}")

    input_data = read_file(file_name)

    walker = parse_challenge(input_data)
    power_walker = parse_challenge(input_data)

    # res_part1 = solve(parsed_input)  # If needed, implement and call solve
    # res_part2 = solve_part2_heavily_optimized(walker)  # If needed, implement and call solve_part2_heavily_optimized
    # res_magic = magic_solve_part2(parsed_input)  # If needed, implement and call magic_solve_part2
    res_power = solve(power_walker)

    # print("Part 1's result is", res_part1)  # Uncomment if needed
    # print("Part 2's result is", res_part2)  # Uncomment if needed
    # print("Part 2's result is", res_magic)  # Uncomment if needed
    print("Part 2's result is", res_power)
